### PR TITLE
More fine grained control of `clear_prefix` and `storage_kill` from the runtime

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: proposals/0000-runtime-environment-clear-prefix-and-storage-kill.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+any information on how many keys got deleted in the state and how got deleted in the overlay.
+
+## Detailed Solution design
+
+Introduce the following new host functions:
+
+```rust
+fn ext_Storage_clear_prefix_version_3(prefix: u64, maybe_limit: u32, maybe_cursor: u64) -> u64;
+fn ext_DefaultChildStorage_clear_prefix_version_3(storage_key: u64, prefix: u64, maybe_limit: u32, maybe_cursor: u64) -> u64;
+fn ext_DefaultChildStorage_storage_kill_version_4(storage_key: u64, maybe_limit: u32, maybe_cursor: u64) -> u64;
```
Comment (tomaka) [inline proposals/0000-runtime-environment-clear-prefix-and-storage-kill.md]: Is there a reason for the CamelCasing? All the other host functions are snake_case

**Comment by tomaka** [comment]
This doesn't really explain why the change is needed or why this design was chosen

File: proposals/0000-runtime-environment-clear-prefix-and-storage-kill.md (hunk 1 above)
Comment (bkchr) [reply proposals/0000-runtime-environment-clear-prefix-and-storage-kill.md]: Yeah, I trusted my own docs :P I will correct it, snake case is correct.

**Comment by bkchr** [comment]
> This doesn't really explain why the change is needed or why this design was chosen

Extended the motivation.

**Comment by Noc2** [comment]
We will archive this repository in favor of the new [RFCs repository by the Fellowship](https://github.com/polkadot-fellows/RFCs). So, I'm closing the issue here. I recommend opening a PR at the new repository.

**Pull request closed without merging**